ocrd_tool.schema.yml

```
@@ -126,3 +126,52 @@ properties:
                 - layout/segmentation/word
                 - layout/segmentation/classification
                 - layout/analysis
+          resource_locations:
+            type: array
+            description: The locations in the filesystem this processor supports for resource lookup
+            default: ['data', 'cwd', 'system']
+            items:
+              type: string
+              enum: ['data', 'cwd', 'system']
+          resources:
+            type: array
+            description: Resources for this processor
+            items:
+              type: object
+              additionalProperties: false
+              required:
+                - url
+                - description
+                - name
+                - size
+              properties:
+                url:
+                  type: string
+                  description: URLs of all components of this resource
+                description:
+                  type: string
+                  description: A description of the resource
+                name:
+                  type: string
+                  description: Name to store the resource as
+                type:
+                  type: string
+                  enum: ['file', 'github-dir', 'tarball']
```

```suggestion
                  enum: ['file', 'directory', 'archive']
```

Should be renamed in existing implementation (`resource_manager.py`, `resource_list.yml`), too.

---

`github-dir` is only a stub in `resource_manager.py`, not implemented. Maybe we can decide on how to move forward with directory resources to replace `github-dir` with `directory`. Then we probably should also distinguish between files and directories within an archive. So far, we only have directories in archives (calamari, eynollah etc) but this might change in the future.

---

> Then we probably should also distinguish between files and directories within an archive. So far, we only have directories in archives (calamari, eynollah etc) but this might change in the future.

We have `path_in_archive` though. So even if you only target a file in an archive, it should already be covered.